src/main/kotlin/org/rust/ide/statistics/RsCodeVisionUsageCollector.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Use of this source code is governed by the MIT license that can be
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.statistics
+
+import com.intellij.internal.statistic.eventLog.EventLogGroup
+import com.intellij.internal.statistic.eventLog.events.EventFields
+import com.intellij.internal.statistic.eventLog.events.StringEventField
+import com.intellij.internal.statistic.service.fus.collectors.CounterUsagesCollector
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.RsMacroDefinitionBase
+
+@Suppress("UnstableApiUsage")
+class RsCodeVisionUsageCollector : CounterUsagesCollector() {
+    override fun getGroup(): EventLogGroup = GROUP
+
+    companion object {
+        fun logAuthorClicked(element: PsiElement) {
+            val location = getLocation(element)
+            AUTHOR_CLICKED_EVENT.log(element.project, location)
+        }
+
+        fun logUsagesClicked(element: PsiElement) {
+            val location = getLocation(element)
+            USAGES_CLICKED_EVENT_ID.log(element.project, location)
+        }
+
+        private fun getLocation(element: PsiElement): String {
+            return when (element) {
+                is RsFunction -> Location.FUNCTION
+                is RsStructItem -> Location.STRUCT_ITEM
+                is RsEnumItem -> Location.ENUM_ITEM
+                is RsEnumVariant -> Location.ENUM_VARIANT
+                is RsNamedFieldDecl -> Location.NAMED_FIELD_DECL
+                is RsTraitItem -> Location.TRAIT_ITEM
+                is RsImplItem -> Location.IMPL_ITEM
+                is RsTypeAlias -> Location.TYPE_ALIAS
+                is RsConstant -> Location.CONSTANT
+                is RsMacroDefinitionBase -> Location.MACRO_DEF
+                is RsModItem -> Location.MOD_ITEM
+                else -> Location.UNKNOWN
+            }.name
+        }
+
+        private val GROUP = EventLogGroup("rust.code.vision", 1)
+
+        private enum class Location(name: String) {
```

Comment:
`name` is unused. You just pass some string values and don't keep them anywhere. Yeah, in `getLocation ` you use `name` property, but it's builtin enum property